great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+        limit_mode: Optional[Union[CardinalityLimitMode, str]] = None,
+        max_unique_values: Optional[int] = None,
+        max_proportion_unique: Optional[int] = None,
+        exclude_columns: Optional[List[str]] = None,
+    ):
+        """Create column domains where cardinality is within the specified limit.
+
+        Cardinality refers to the number of unique values in a given domain.
+        Categorical generally refers to columns with relatively limited
+        number of unique values.
+        Limit mode can be absolute (number of unique values) or relative
+        (proportion of unique values). You can choose one of: limit_mode,
+        max_unique_values or max_proportion_unique to specify the cardinality
+        limit.
+        Note that the limit must be met for each batch separately that is
+        supplied in the batch_request or the column domain will not be included.
+        Note that the columns used will be from the first batch retrieved
+        via the batch_request. If other batches contain additional columns,
+        these will not be considered.
+
+        Args:
+            data_context: DataContext associated with this profiler.
+            batch_request: BatchRequest to be optionally used to define batches
+                to consider for this domain builder.
+            limit_mode: CardinalityLimitMode or string name of the mode
+                defining the maximum allowable cardinality to use when
+                filtering columns.
+            max_unique_values: number of max unique rows for a custom
+                cardinality limit to use when filtering columns.
+            max_proportion_unique: proportion of unique values for a
+                custom cardinality limit to use when filtering columns.
+            exclude_columns: If provided, these columns are pre-filtered and
+                excluded from consideration, cardinality is not computed.
+        """
+
+        super().__init__(
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._cardinality_checker = CardinalityChecker(
+            limit_mode=limit_mode,
+            max_unique_values=max_unique_values,
+            max_proportion_unique=max_proportion_unique,
+        )
+
+        self._exclude_columns = exclude_columns
+
+    @property
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        return MetricDomainTypes.COLUMN
+
+    def _get_domains(
+        self,
+        variables: Optional[ParameterContainer] = None,
+    ) -> List[Domain]:
+        """Return domains matching the selected limit_mode.
+
+        Args:
+            variables: Optional variables to substitute when evaluating.
+
+        Returns:
+            List of domains that match the desired cardinality.
+        """
+
+        batch_ids: List[str] = self._get_batch_ids(variables=variables)
```

Comment:
@anthonyburdi I did not envision `get_batch_ids()` for `DomainBuilder` and hence made it private (as an internal-use-only method).  But your use case indicates that it should be made public -- might it make sense then to change `_get_batch_ids()` to `get_batch_ids()` throughout?  If you agree, then could you please affect the change?  Thanks!

Reply:
Done!